worker-message-prioritization-target-capacity-calculators/src/main/java/com/github/workerframework/workermessageprioritization/targetcapacitycalculators/K8sTargetQueueCapacityProvider.java

```diff
@@ -0,0 +1,164 @@
+                kubernetesLabelCacheExpiryMinutes);
+    }
+
+    @Override
+    public long get(final Queue targetQueue)
+    {
+        try {
+            return targetQueueToMaxLengthCache.get(targetQueue);
+        } catch (final ExecutionException executionException) {
+            LOGGER.error(String.format("Cannot get max length for the %s queue as an ExecutionException was thrown. " +
+                    "Falling back to using a max length of %s", targetQueue, TARGET_QUEUE_MAX_LENGTH_FALLBACK), executionException);
+
+            return TARGET_QUEUE_MAX_LENGTH_FALLBACK;
+        }
+    }
+
+    private long getTargetMaxQueueLengthFromKubernetes(final Queue targetQueue)
+    {
+        // Get the target queue name
+        final String targetQueueName = targetQueue.getName();
+
+        // Loop through all provided namespaces
+        for (final String kubernetesNamespace : kubernetesNamespaces) {
+
+            try {
+                // Loop through all deployments (e.g. "elastic-worker", "elastic-query-worker")
+                for (final V1Deployment deployment : Kubectl.get(V1Deployment.class).namespace(kubernetesNamespace).execute()) {
+
+                    // Get the metadata
+                    final V1ObjectMeta metadata = deployment.getMetadata();
+                    if (metadata == null) {
+                        continue;
+                    }
+
+                    // Get the labels from the metadata
+                    final Map<String, String> labels = metadata.getLabels();
+                    if (labels == null) {
+                        continue;
+                    }
+
+                    // Check if there is a target queue name label
+                    if (!labels.containsKey(MESSAGE_PRIORITIZATION_TARGET_QUEUE_NAME_LABEL)) {
+                        continue;
+                    }
+
+                    // Check if the target queue name label value matches the target queue name provided to this method
+                    if (!labels.get(MESSAGE_PRIORITIZATION_TARGET_QUEUE_NAME_LABEL).equals(targetQueueName)) {
+                        continue;
+                    }
+
+                    // Check if there is a target queue max length label
+                    if (!labels.containsKey(MESSAGE_PRIORITIZATION_TARGET_QUEUE_MAX_LENGTH_LABEL)) {
+                        // Throw RuntimeException as this indicates a deployment error that should never happen in production
+                        throw new RuntimeException(String.format(
+                                "Cannot get max length for the %s queue. The %s worker is missing the %s label",
+                                targetQueueName, metadata.getName(), MESSAGE_PRIORITIZATION_TARGET_QUEUE_MAX_LENGTH_LABEL));
+                    }
+
+                    // Return the value of the target queue max length label
+                    final Long targetQueueMaxLength = Long.valueOf(labels.get(MESSAGE_PRIORITIZATION_TARGET_QUEUE_MAX_LENGTH_LABEL));
```

Comment:
You can use Long.parseLong here to get to the primitive long, returned by the method, avoiding using the wrapper Long